common/scala/src/main/scala/whisk/core/connector/Message.scala

```diff
@@ -118,3 +118,21 @@ object PingMessage extends DefaultJsonProtocol {
   def parse(msg: String) = Try(serdes.read(msg.parseJson))
   implicit val serdes = jsonFormat(PingMessage.apply _, "name")
 }
+
+case class InvokerIdRequestMessage(instanceName: String, proposedId: Int) extends Message {
```

Comment:
As indicated below, `proposedId` could be `Option[Int]` to model possible absence of a proposal.